### PR TITLE
set default value for repeating group count

### PR DESCRIPTION
**Pull request by syedowaisali90**
Hi @vincent-karuri I have created a PR for review

### CODE REVIEW

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/RepeatingGroupFactory.java [hunk 1]
```diff
@@ -152,6 +152,7 @@ private JSONObject getRepeatingGroupCountObj(WidgetArgs widgetArgs) throws JSONE
         repeatingGroupCountObj.put(OPENMRS_ENTITY_ID, "");
         repeatingGroupCountObj.put(TYPE, "");
         repeatingGroupCountObj.put(TEXT, widgetArgs.getJsonObject().get(REFERENCE_EDIT_TEXT_HINT));
+        repeatingGroupCountObj.put(VALUE, "0");
```
Comment (vincent-karuri) [inline android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/RepeatingGroupFactory.java:155]: Could you add a test to verify that all fields of the `repeatingGroupCountObj` are correctly populated?

Comment (vincent-karuri) [reply android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/RepeatingGroupFactory.java:155]: And in particular, to verify that the value is 0 by default and that it is always reset to zero when text is cleared.

Comment (syedowaisali90) [reply android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/RepeatingGroupFactory.java:155]: done

File: android-json-form-wizard/src/test/java/com/vijay/jsonwizard/widgets/RepeatingGroupFactoryTest.java [hunk 2]
```diff
@@ -180,6 +180,17 @@ public void testUniqueChildElementKeyGenerationShouldContainChildKeyAsComponent(
...
     }
 
+    @Test
+    public void testRepeatingGroupDefaultCount() throws Exception {
+        View rootLayout = invokeGetViewsFromJson().get(0);
+
+        JSONObject repeatingGroupCountObj = step.getJSONArray(JsonFormConstants.FIELDS).getJSONObject(0);
+        Assert.assertEquals("0", repeatingGroupCountObj.getString(JsonFormConstants.VALUE));
+
+        ((MaterialEditText) rootLayout.findViewById(R.id.reference_edit_text)).setText("");
```
Comment (vincent-karuri) [inline android-json-form-wizard/src/test/java/com/vijay/jsonwizard/widgets/RepeatingGroupFactoryTest.java:190]: Could you add an intermediate step where you set the text to something else and verify that that value is updated on `repeatingGroupCountObj`,  then set it to blank and verify that it's reset to `0`?

Comment (syedowaisali90) [reply android-json-form-wizard/src/test/java/com/vijay/jsonwizard/widgets/RepeatingGroupFactoryTest.java:193]: added.

**Review by vincent-karuri — APPROVED** [review]

**Pull request merged**